clang/test/CIR/CodeGen/address-space.c

```diff
@@ -0,0 +1,12 @@
+// RUN: %clang_cc1 -triple x86_64-unknown-linux-gnu -fclangir -emit-cir %s -o %t.cir
+// RUN: FileCheck --input-file=%t.cir %s
+
+// CHECK: cir.func {{@.*foo.*}}(%arg0: !cir.ptr<!s32i>
+void foo(int __attribute__((address_space(0))) *arg) {
+  return;
+}
```

Comment:
This is a great test, but do notice that it tests two things:
1. Being able to support address space attribute in the front-end
2. Not printing the addressspace in case it is "0".

I would suggest to add a specific test for the second one in `clang/test/CIR/IR`.

Reply:
> I would suggest to add a specific test for the second one in `clang/test/CIR/IR`.

Agreed, both here and there would be great

Reply:
I'd also prefer if you add the LLVMs check here, see other tests for examples. No need to change `clang/test/CIR/Lowering/address-space.cir` though, that's also good.